### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by maybegreat48**
* The invoker has been rewritten to be more efficient and thread-safe
* Zydis has been replaced by the Hacker Disassembler Engine found in MinHook
* Compiler optimizations have been enabled, halving DLL size
* Refactored hooking code into a new directory
* Bug fixes

### CODE REVIEW

File: src/lua/natives/__pycache__/natives_gen.cpython-311.pyc
Comment (Yimura) [inline src/lua/natives/__pycache__/natives_gen.cpython-311.pyc:1]: Please remove `__pycache__` folder and add it to `.gitignore`.

File: src/native_hooks/tunables.hpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace big
 
 		inline void NETWORK_ACCESS_TUNABLE_BOOL_MODIFICATION_DETECTION_REGISTRATION_HASH(rage::scrNativeCallContext* src)
 		{
-			g_tunables_service->register_tunable(src->get_arg<Hash>(1), src->get_arg<BOOL*>(2));
+			g_tunables_service->register_tunable(src->get_arg<Hash>(1), src->get_arg<bool*>(2));
```
Comment (Yimura) [inline src/native_hooks/tunables.hpp:24]: Intentional change from `BOOL` to `bool`?

**Review by Yimura — CHANGES_REQUESTED** [review]

**Comment by Yimura** [comment]
Any chance we can rename the `scripts/` folder to `cmake/` and start moving some python helper scripts into a new `scripts/` folder?

I have been planning on doing this but haven't made to time to get around to it.

File: src/lua/natives/__pycache__/natives_gen.cpython-311.pyc
Comment (maybegreat48) [reply src/lua/natives/__pycache__/natives_gen.cpython-311.pyc:1]: VS didn't show me that for some reason

File: src/native_hooks/tunables.hpp (hunk 1 above)
Comment (maybegreat48) [reply src/native_hooks/tunables.hpp:24]: No, but it shouldn't matter

**Comment by xiaoxiao921** [comment]
Very confused about the release with deb info stuff, do we even want all of this? Why don't we just remove all the info and just use a good classic release mode? Also the nightly never used it also unless it I'm missing something

**Comment by Yimura** [comment]
> Very confused about the release with deb info stuff, do we even want all of this? Why don't we just remove all the info and just use a good classic release mode? Also the nightly never used it also unless it I'm missing something

He probably wants users to start taking the pdb as well so the crash dumps are easier to decipher.

**Comment by xiaoxiao921** [comment]
https://github.com/return-of-modding/ReturnOfModding/commit/aff7004aec33aea67e00dd72bf84d6bec760bd2d

**Review by Yimura — APPROVED** [review]
Fine by me to merge it now, I or someone else can do the change in regards to the scripts/ directory.

**Pull request merged**